packages/rest/src/types.ts

```diff
@@ -4,11 +4,28 @@
 // License text available at https://opensource.org/licenses/MIT
 
 import {Binding, BoundValue} from '@loopback/context';
+import {Component as CoreComponent} from '@loopback/core';
 import {ResolvedRoute, RouteEntry} from './router/routing-table';
 import {Request, Response} from 'express';
 
 export {Request, Response};
 
+/**
+ * Static assets settings
+ */
+export interface StaticAssetsConfig {
+  path: string;
+  rootDir: string;
+  options?: object;
+}
+
+export interface Component extends CoreComponent {
```

Comment:
It's probably better to name it as `ComponentWithStaticAssets`.

Reply:
We my want to add additional exports besides static assets later in the future. I think `RestComponent` would be a better name?

I also think the interface should not be inheriting from `CoreComponent` to allow composition of multiple transport-specific interfaces, e.g. `Component & RestComponent & GrapQLComponent`.

Reply:
Don't we have a class named as `RestComponent`?

Reply:
> Don't we have a class named as RestComponent
@raymondfeng Yes, we have that already. For now I have renamed it to `StaticComponent`.

> I also think the interface should not be inheriting from CoreComponent ...

@bajtos in that case we'll have to do `componentCtor: Constructor<Component | StaticComponent>`, etc. Is that OK?

I think this should be an independent interface from `RestComponent`, `StaticComponent` does not need any of the properties defined in `RestComponent`. Clubbing them together is not efficient.

Reply:
> Don't we have a class named as RestComponent?

We actually do! It's a component adding REST servers to an app. 

The intention here was to describe a component that's contributing REST-specific artifacts like static assets.

What a confusion 🙈